wasm/src/main/java/com/dylibso/chicory/wasm/WasmWriter.java

```diff
@@ -0,0 +1,44 @@
+package com.dylibso.chicory.wasm;
+
+import static java.lang.Integer.toUnsignedLong;
+
+import com.dylibso.chicory.wasm.types.RawSection;
+import java.io.ByteArrayOutputStream;
+
+public final class WasmWriter {
+    private static final byte[] MAGIC = {0x00, 0x61, 0x73, 0x6D};
```

Comment:
I think we should re-use the same constants as the Parser?
https://github.com/dylibso/chicory/blob/cad19fef9b96179e76bc0f795e390a6f6e8aa662/wasm/src/main/java/com/dylibso/chicory/wasm/Parser.java#L80

or have a separate `Constants` file in the module

Reply:
If you're fine to change the parser to use a byte array. I copied the hex format from the specification.

Reply:
Totally fine with it :+1: any downside I haven't considered?

Reply:
Done